internal/app/versionResolver.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: This file has the package app.
+
+package app
+
+import (
+	"fmt"
+
+	"github.com/getoutreach/devenv/internal/app/resolvers/github"
+	"github.com/getoutreach/devenv/internal/app/resolvers/maestro"
+	"github.com/getoutreach/devenv/internal/app/resolvers/mock"
+	"github.com/getoutreach/devenv/internal/app/resolvers/types"
+	"github.com/getoutreach/gobox/pkg/box"
+)
+
+const (
+	// MaestroVersionResolverString is the VersionResolverType type for maestro
+	MaestroVersionResolverString types.VersionResolverName = "maestro"
+
+	// MaestroVersionResolverString is the VersionResolverType type for github
+	GitHubVersionResolverString types.VersionResolverName = "github"
+
+	// VersionResolverMockString used to mock version resolver when testing
+	VersionResolverMockString types.VersionResolverName = "mockVersionResolver"
+)
+
+// StringToVersionResolverConfigMap converts a VersionResolverType to a
+// VersionResolver config if found. Returns an error if not found
```

Comment:
```suggestion
// VersionResolver config if found.
```

Maps don't return errors if a key doesn't exist, so slight adjustment there.